SingularityService/src/main/java/com/hubspot/singularity/data/zkmigrations/NamespaceActiveTasksMigration.java

```diff
@@ -0,0 +1,47 @@
+package com.hubspot.singularity.data.zkmigrations;
+
+import java.util.List;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.utils.ZKPaths;
+
+import com.google.inject.Inject;
+import com.hubspot.singularity.SingularityTaskId;
+
+public class NamespaceActiveTasksMigration extends ZkDataMigration {
+  private static final String ACTIVE_TASKS_ROOT = "/tasks/active";
+  private static final String ACTIVE_STATUSES_ROOT = "/tasks/statuses";
+
+  private final CuratorFramework curatorFramework;
+
+  @Inject
+  public NamespaceActiveTasksMigration(CuratorFramework curatorFramework) {
+    super(14);
+    this.curatorFramework = curatorFramework;
+  }
+
+  @Override
+  public void applyMigration() {
+    try {
+      if (curatorFramework.checkExists().forPath(ACTIVE_TASKS_ROOT) != null) {
+        List<String> currentActive = curatorFramework.getChildren().forPath(ACTIVE_STATUSES_ROOT);
+        for (String taskIdString : currentActive) {
+          SingularityTaskId taskId = SingularityTaskId.valueOf(taskIdString);
+          String oldPath = ZKPaths.makePath(ACTIVE_STATUSES_ROOT, taskIdString);
```

Comment:
We're running this code based on whether `ACTIVE_TASKS_ROOT` exists in ZK, but we're never actually reading from that path during this migration. Just double-checking, is that intended?

Reply:
We remove it as the last step of the migration which is why it's my initial check. There isn't actually any data in those nodes, they are only an index we call getChildren on. The /tasks/statuses is where the actual data is and what I am namespacing. However, realizing that if this died half way though, some of the children on the next run would be request ids instead. Going to add a fix so this will be able to run on a half-migrated dataset as well